docs/rules/id-blacklist.md

```diff
@@ -0,0 +1,66 @@
+# Blacklist certain identifiers to prevent them being used (id-blacklist)
+
+> "There are only two hard things in Computer Science: cache invalidation and naming things." — Phil Karlton
+
+Bad names can lead to hard to decipher code. Using generic names, such as `data` don't infer much about the code and the values it receives. This rule allows you to configure a blacklist of bad identifier names, that you don't want to see in your code.
+
+## Rule Details
+
+This rule compares assignments and function definitions to a provided list of identifier names. If the identifier is present in the list, it will return an error.
+It doesn't apply to function calls, so that you can still use functions or objects you do not have control over.
```

Comment:
Can you add a bulleted list of locations where the check is done? I'm not sure it's obvious from the examples or that people will understand, in general, where an identifier is used.